engine/src/main/java/org/terasology/engine/persistence/serializers/PrefabSerializer.java

```diff
@@ -155,7 +155,8 @@ private void applyComponentChanges(Module context, EntityData.Prefab prefabData,
                     }
                 }
             } else if (componentData.hasType()) {
-                logger.warn("Prefab '{}' contains unknown component '{}'", prefabData.getName(), componentData.getType());
+                logger.atWarn().addArgument(() -> prefabData.getName()).addArgument(() -> componentData.getType()).
+                        log("Prefab '{}' contains unknown component '{}'");
```

Comment:
`getName()` is simple getter method and `getType()` should be O(1) so impact neglectable
```suggestion
                logger.warn("Prefab '{}' contains unknown component '{}'", prefabData.getName(), componentData.getType()); //NOPMD
```